### PR TITLE
[Improvement](pipeline) blocking tasks before putting in runnable queue

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

File: be/src/pipeline/dependency.h [hunk 1]
```diff
@@ -84,24 +83,27 @@ struct BasicSharedState {
     Dependency* create_sink_dependency(int dest_id, int node_id, std::string name);
 };
 
+/**
+ * We divide dependency into three categories:
+ * 1. BEFORE_EXECUTION: pipeline tasks need to wait for this before execution. (i.e. ExecutionDependency, RuntimeFilterDependency)
+ * 2. DURING_EXECUTION: In execution phase, tasks should be running only if this dependency is ready. (i.e. Read/Write Dependency)
+ * 3. AFTER_EXECUTION: After execution phase, tasks should be closed after this dependency is ready.
+ */
+enum class DependencyType : uint8_t {
```
Comment (yiguolei) [inline be/src/pipeline/dependency.h:92]: We should not add dependency type.
It is very difficult for the developer to use. Many developer will not know the type related with wake up method.

**Review by yiguolei — CHANGES_REQUESTED** [review]
I will think about it.

**Pull request closed without merging**